pkg/cassandra/config.go

```diff
@@ -103,6 +103,40 @@ func newConfig(apiConfig api.CassandraConfig, cassandraVersion string) (config,
 	return cfg, nil
 }
 
+func addEncryptionOptions(apiConfig *api.CassandraConfig, encryptionStoresSecrets EncryptionStoresPasswords, cassandraVersion string) api.CassandraConfig {
+	updatedConfig := apiConfig.DeepCopy()
+	if updatedConfig.CassandraYaml.ClientEncryptionOptions == nil && updatedConfig.CassandraYaml.ServerEncryptionOptions == nil {
+		return *updatedConfig
+	}
+
+	if updatedConfig.CassandraYaml.ClientEncryptionOptions != nil {
+		if updatedConfig.CassandraYaml.ClientEncryptionOptions.Enabled {
+			keystorePath := fmt.Sprintf("%s/%s", StoreMountFullPath("client", "keystore"), "keystore")
+			truststorePath := fmt.Sprintf("%s/%s", StoreMountFullPath("client", "truststore"), "truststore")
+			updatedConfig.CassandraYaml.ClientEncryptionOptions.Keystore = pointer.String(keystorePath)
+			updatedConfig.CassandraYaml.ClientEncryptionOptions.Truststore = pointer.String(truststorePath)
+			updatedConfig.CassandraYaml.ClientEncryptionOptions.KeystorePassword = &encryptionStoresSecrets.ClientKeystorePassword
+			updatedConfig.CassandraYaml.ClientEncryptionOptions.TruststorePassword = &encryptionStoresSecrets.ClientTruststorePassword
+		}
+	}
+	if updatedConfig.CassandraYaml.ServerEncryptionOptions != nil {
+		if *updatedConfig.CassandraYaml.ServerEncryptionOptions.Enabled {
```

Comment:
Beware that you could have a nil pointer dereference here if `ServerEncryptionOptions.Enabled` is nil – which is why I advocate for creating a dedicated method `IsEnabled()`.